File: backend/app/schemas/lecture.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class LectureCreate(BaseModel):
    title: str
    video_name: str


class LectureRead(BaseModel):
    lecture_id: str
    title: str
    video_name: str
    status: str
    teacher_id: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None
    video_url: Optional[str] = None
    cloudinary_public_id: Optional[str] = None

    model_config = {"from_attributes": True}

File: frontend/types/lecture.ts
export interface Lecture {
  lecture_id: string
  title: string
  video_name: string
  status: 'live' | 'available' | 'completed' | string
  teacher_id?: string | null
  teacher_name?: string | null
  created_at?: string
  completed_at?: string | null
  video_url?: string | null
  cloudinary_public_id?: string | null
}

export interface LectureCreate {
  title: string
  video_name?: string
}

export interface LectureEvent {
  event_id: string
  lecture_id: string
  timestamp: number
  type: string
  source: string
  content: string
  metadata: Record<string, unknown>
}

export type LectureStatus = 'idle' | 'starting' | 'live' | 'completed' | 'error' | string
